controller/oidc_auth/negotiate.go

```diff
@@ -12,26 +30,47 @@ import (
 // cannot be satisfied an error is returned.
 func negotiateResponseContentType(r *http.Request) (string, *errorz.ApiError) {
 	acceptHeader := r.Header.Get(AcceptHeader)
+
+	if acceptHeader == "" {
+		return JsonContentType, nil
+	}
+
 	contentTypes := parseAcceptHeader(acceptHeader)
 
-	if len(contentTypes) == 0 || acceptHeader == "" {
+	if len(contentTypes) == 0 {
 		return JsonContentType, nil
 	}
 
 	for _, contentType := range contentTypes {
-		if contentType == JsonContentType {
-			return contentType, nil
-		} else if contentType == HtmlContentType {
+		if contentType.Match(JsonContentType) {
+			return JsonContentType, nil
+		} else if contentType.Match(HtmlContentType) {
 			return HtmlContentType, nil
 		}
 	}
 
 	return "", newNotAcceptableError(acceptHeader)
 }
 
+// parseContentType extracts the media type from a Content-Type header value
+func parseContentTypeMediaType(contentType string) string {
+	contentType = strings.TrimSpace(contentType)
+
+	if contentType == "" {
```

Comment:
this if doesn't seem necessary unless you're trying to optimize the function?

Reply:
Stylistic, it is a guard statement. Simply makes the condition explicit.